### ISSUE TITLE
Not reading linkname and ustar

### ISSUE THREAD

**Issue by PrzemyslawBanasiak** [InvokIT/js-untar#6]
There is a bug in code with reading linkname and ustar. I have already debugged it:
In file untar-worker.js, line 152 you have `file.linkname = stream.readString(1)`, but linkname length should be 100. Additionally it turns out that changing that doesn't fully fix the problem with ustar. When reading six characters for ustarFormat there is a space after it, which is not present in if condition. You can fix it be comparing to `"ustar "` or using `indexOf`. You could test it against your own files but it reproduced in all files I already tested.

**Comment by Qvazar** [comment]
Thanks for spotting that typo, linkname is of course 100 chars long. :+1: 

The tar spec says that the "ustar"-string should be `ustar` followed by a `NUL`-character, not a `space`. The `NUL`-char is ignored in `readString()`, so `"ustar"` is returned. Since your tar-file has a space instead of a `NUL`, where does your tar-file come from?

The library should of course still work, so I implemented your `indexOf`-suggestion.

Fixed in 0.1.3
